src/tenant/admin.py

```diff
@@ -115,6 +129,102 @@ def get_queryset(self, request):
                     tenant.update_cached_fields()
         return qs
 
+    @admin.action(description="Send an email message to all owners for the selected tenant(s)")
+    def message_selected(self, request, queryset):
+        """
+        Send an email message to all owners for selected tenant(s).
+
+        This action first displays an intermediate page which shows compose message form and
+        a list of all recipients.
+
+        Next, it send email message to all selected users and redirects back to the change list.
+        """
+        from siteconfig.models import SiteConfig
+
+        def get_owner_or_none():
+            """Returns owner (User) object or None"""
+            return SiteConfig.get().deck_owner or None
+
+        # get a list of selected tenant(s), excluding public schema
+        objects = self.model.objects.filter(
+            pk__in=[str(x) for x in request.POST.getlist(helpers.ACTION_CHECKBOX_NAME)]
+        ).exclude(schema_name=get_public_schema_name())
+
+        # get a list of all owners (recipients) for selected tenant(s)
+        recipient_list = []
+        for tenant in objects:
+            with tenant_context(tenant):
+                owner = get_owner_or_none()
+                if owner is None:  # where is the owner?
+                    continue
+                # get the full name of the user, or if none is supplied will return the username
+                full_name_or_username = owner.get_full_name() or owner.username
+
+                # get the email address, but only primary and verified
+                for primary_email_address in EmailAddress.objects.filter(user=owner, primary=True, verified=True):
+                    # make sure it's primary email for real
+                    if primary_email_address.email == user_email(owner):
+                        email = owner.email
+
+                if len(email):  # skip, if email address is empty
+                    recipient_list.append(f"{full_name_or_username} <{email}>")
```

Comment:
Could you include the tenant in the email? That would be good feedback for the user of the form to confirm the expected decks are included, for example:
f"{tenant.name} - {full_name_or_username} <{email}>"
But please confirm the proper field on the tenant model.